x-pack/plugins/actions/server/lib/action_executor.ts

```diff
@@ -97,64 +118,283 @@ export class ActionExecutor {
   }
 
   public async execute({
+    actionExecutionId,
+    actionId,
+    actionInfo,
+    consumer,
+    executionId,
+    isEphemeral,
+    request,
+    params,
+    relatedSavedObjects,
+    source,
+    taskInfo,
+  }: ExecuteOptions): Promise<ActionTypeExecutorResult<unknown>> {
+    const {
+      actionTypeRegistry,
+      getActionsAuthorizationWithRequest,
+      getServices,
+      security,
+      spaces,
+    } = this.actionExecutorContext!;
+
+    const services = getServices(request);
+    const spaceId = spaces && spaces.getSpaceId(request);
+    const namespace = spaceId && spaceId !== 'default' ? { namespace: spaceId } : {};
+    const authorization = getActionsAuthorizationWithRequest(request);
+    const currentUser = security?.authc.getCurrentUser(request);
+
+    return await this.executeHelper({
+      actionExecutionId,
+      actionId,
+      actionInfo,
+      consumer,
+      currentUser,
+      ensureAuthorizedFn: async (connectorTypeId: string) => {
+        /**
+         * Ensures correct permissions for execution and
+         * performs authorization checks for system actions.
+         * It will thrown an error in case of failure.
+         */
+        await ensureAuthorizedToExecute({
+          params,
+          actionId,
+          actionTypeId: connectorTypeId,
+          actionTypeRegistry,
+          authorization,
+        });
+      },
+      executeLabel: `execute_action`,
+      executionId,
+      isEphemeral,
+      namespace,
+      params,
+      relatedSavedObjects,
+      request,
+      services,
+      source,
+      spaceId,
+      taskInfo,
+    });
+  }
+
+  public async executeUnsecured({
+    actionExecutionId,
     actionId,
     params,
+    relatedSavedObjects,
+    spaceId,
+    source,
+  }: UnsecuredExecuteOptions): Promise<ActionTypeExecutorResult<unknown>> {
+    const { actionTypeRegistry, getUnsecuredServices } = this.actionExecutorContext!;
+
+    const services = getUnsecuredServices();
+    const namespace = spaceId && spaceId !== 'default' ? { namespace: spaceId } : {};
+
+    return await this.executeHelper({
+      actionExecutionId,
+      actionId,
+      ensureAuthorizedFn: async (connectorTypeId: string) => {
+        // We don't allow execute system actions in unsecured manner because they require a request
+        if (actionTypeRegistry.isSystemActionType(connectorTypeId)) {
```

Comment:
Current implementation doesn't allow calling system actions in this manner. We can address in the future if this ability is needed.